Data/Aeson/Types/FromJSON.hs

```diff
@@ -1282,9 +1282,10 @@ instance ( FieldNames f
         \obj -> checkUnknown obj >> recordParseJSON' p obj
         where
             knownFields :: H.HashMap Text ()
-            knownFields = H.fromList $ map (,()) $
-                fieldNames (undefined :: f a)
-                [pack (tagFieldName (sumEncoding opts)) | fromTaggedSum]
+            knownFields = H.fromList $ map ((,()) . pack) $
+                [(tagFieldName (sumEncoding opts)) | fromTaggedSum] <>
+                (fieldLabelModifier opts <$> fieldNames (undefined :: f a) [])
```

Comment:
The `fieldLabelModifier` should not be applied to the "tag" fields. Hence I could not anymore pass the tag fields into the field names reflection.

I had to choose between: A list concatentation (and these can be inefficient) and a union of 2 hashes, one build from the tag fields and one from the reflected field names.

I went with the list concatenation, as I assume that the number of tag fields will usually be lower and appending to a "small" list is quite fast. Hence also putting the "smaller" list on the LHs of `<>`.

Reply:
Sounds good!